src/infra/AnalyticsReporter.ts

```diff
@@ -4,11 +4,15 @@ import { HttpRequesterService } from '../services/HttpRequesterService';
 import { AnalyticsEvent } from '../models/events/AnalyticsEvent';
 import { BeaconPayload } from '../models/BeaconPayload';
 import { AnalyticsResponse } from '../models/AnalyticsResponse';
+import { Visitor } from '../models/Visitor';
 
 const defaultDomain = 'https://answers.yext-pixel.com';
 
 export class AnalyticsReporter implements AnalyticsService {
-  constructor(private config: AnalyticsConfig, private httpRequesterService: HttpRequesterService) {}
+  private _visitor: Visitor | undefined;
```

Comment:
I think we normally don't include the underscore prefix for typescript private variables but I don't mind having it honestly